### PR TITLE
test_link: refactor code to avoid instability

### PR DESCRIPTION
**Pull request by weia5-emc**

### CODE REVIEW

**Comment by willwu1** [comment]
What causes instability， guest-agent ？

**Comment by weia5-emc** [comment]
> What causes instability， guest-agent ？

I guess it is. After all, it depends on external resource which we can't guarantee. 
This test can be accomplished based on its own resource only, so i changed it.

File: test/functional/test_link_change_error_inject.py [hunk 1]
```diff
@@ -6,53 +6,56 @@
...
+            ]
+        },
         {
-            "outside": 2222,
-            "inside": 22,
-            "protocal": "tcp"
+            "device": "e1000",
+            "id": "e1000.1",
+            "network_mode": "nat",
+            "mac": target_mac,
```
Comment (XiaowenJiang) [inline test/functional/test_link_change_error_inject.py]: good format

File: test/functional/test_link_change_error_inject.py [hunk 2]
```diff
@@ -169,71 +124,51 @@ def tearDownClass(cls):
...
+                "name": "e1000.1",
                 "up": False
             }
         }
-        status_code_error2 = {
+        cmd_nic_up = {
             "execute": "set_link",
             "arguments": {
-                "name": "e1000.0",
+                "name": "e1000.1",
```
Comment (XiaowenJiang) [inline test/functional/test_link_change_error_inject.py]: what is the difference between e1000.0 and e1000.1?

Comment (weia5-emc) [reply test/functional/test_link_change_error_inject.py]: e1000.1 is the nic which i want to operate.
e1000.0 is the nic provides ssh tunnel.

Comment (emcinfrasim) [reply test/functional/test_link_change_error_inject.py]: ok.

File: test/functional/test_link_change_error_inject.py [hunk 3]
```diff
@@ -85,77 +86,31 @@ def start_node():
...
 
 def stop_node():
-    global s
-    global sg
     global conf
+    if conf is None:
+        return
     s.close()
-    sg.close()
+    ssh_close(ssh)
```
Comment (emcinfrasim) [inline test/functional/test_link_change_error_inject.py]: good usage

**Pull request merged**